GH-3897: Deprecate `ChannelSecurityInterceptor`

artembilan opened the pull request:
Fixes https://github.com/spring-projects/spring-integration/issues/3897

Spring Security has deprecated `AccessDecisionManager` and all its infrastructure in favor of `AuthorizationManager`

* Deprecate and AOP `ChannelSecurityInterceptor` and all its infrastructure, including `@SecuredChannel` and respective XML configuration. The `AuthorizationChannelInterceptor` added to respective channels for security or configured as a global channel interceptor fully covers the previous AOP configuration
* Fix deprecation warnings in other tests with security

<!--
Thanks for contributing to Spring Integration. 
Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #) or StackOverflow questions.

See the [Contributor Guidelines for more information](https://github.com/spring-projects/spring-integration/blob/main/CONTRIBUTING.adoc).
-->

artembilan committed:
GH-3897: Deprecate `ChannelSecurityInterceptor`

Fixes https://github.com/spring-projects/spring-integration/issues/3897

Spring Security has deprecated `AccessDecisionManager` and all its infrastructure
in favor of `AuthorizationManager`

* Deprecate and AOP `ChannelSecurityInterceptor` and all its infrastructure,
including `@SecuredChannel` and respective XML configuration.
The `AuthorizationChannelInterceptor` added to respective channels for security
or configured as a global channel interceptor fully covers the previous AOP configuration
* Fix deprecation warnings in other tests with security

garyrussell commented on src/reference/asciidoc/security.adoc:
```diff
@@ -7,6 +7,10 @@ Messaging independence and loose coupling let target systems communicate with ea
 We can either trust all those messages or secure our service against "`infecting`" messages.
 
 Spring Integration, together with https://projects.spring.io/spring-security/[Spring Security], provides a simple and comprehensive way to secure message channels, as well as other part of the integration solution.
+Starting with version 6.0, the `ChannelSecurityInterceptor` as well as its configuration via `@SecuredChannel` annotation and  XML `<secured-channels>` have been deprecation in favor of `AuthorizationChannelInterceptor` from `spring-security-messaging` module.
```
```suggestion
Starting with version 6.0, the `ChannelSecurityInterceptor` as well as its configuration via `@SecuredChannel` annotation and  XML `<secured-channels>` have been deprecation in favor of using `AuthorizationChannelInterceptor` from the `spring-security-messaging` module.
```

garyrussell commented on src/reference/asciidoc/security.adoc:
```diff
@@ -7,6 +7,10 @@ Messaging independence and loose coupling let target systems communicate with ea
 
 Spring Integration, together with https://projects.spring.io/spring-security/[Spring Security], provides a simple and comprehensive way to secure message channels, as well as other part of the integration solution.
+Starting with version 6.0, the `ChannelSecurityInterceptor` as well as its configuration via `@SecuredChannel` annotation and  XML `<secured-channels>` have been deprecation in favor of `AuthorizationChannelInterceptor` from `spring-security-messaging` module.
+The respective `AuthorizationManager` infrastructure fully covers previously supported role-based authentication, plus it allows to configure any other possible authorization strategies including.
```
```suggestion
The respective `AuthorizationManager` infrastructure fully covers the previously supported role-based authentication, plus it allows the configuration of any other possible authorization strategies.
```

garyrussell reviewed (changes requested):
Just doc polishing.

artembilan committed:
Fix language in docs

Co-authored-by: Gary Russell <grussell@vmware.com>

garyrussell commented:
Needs rebase/conflicts.

artembilan commented:
Looks like there are more deprecation warnings.

Do not merge yet.

artembilan committed:
* Remove `forRemoval` attr from `@Deprecated` markers for Security classes:
looks like to mark `@Deprecated` and even `@SuppressWarnings("deprecation")`
don't silence warnings on compilation

garyrussell commented:
>221 tests completed, 78 failed, 1 skipped

artembilan commented:
@garyrussell ,

Looks like that one was an interim failure.
It is good now.
I guess the problem was related to the release process we have these days.

garyrussell commented:
Conflict in build.grade.

garyrussell committed:
Merge branch 'main' into GH-3897
